j5/backends/console/arduino/uno.py

```diff
@@ -0,0 +1,113 @@
+"""Console Backend for the Arduino Uno."""
+from typing import Mapping, Optional, Set, Type
+
+from j5.backends import Backend
+from j5.backends.console import Console, ConsoleEnvironment
+from j5.boards import Board
+from j5.boards.arduino import ArduinoUnoBoard
+from j5.components import GPIOPinInterface, GPIOPinMode, LEDInterface
+
+
+class PinData:
```

Comment:
Unfortunately this is not possible afaik.

`NamedTuple` has immutable properties and we need to modify them in this Backend.

An alternative is [`recordclass`](https://pypi.python.org/pypi/recordclass), but this would add another dependency, so I'd rather use a class.

Reply:
I think [`namedtuple._replace`](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace) is the canonical way to "modify" a named tuple, but that's probably an uglier solution. Lets leave this as it is.